Objects/dictobject.c

```diff
@@ -5476,7 +5476,9 @@ _PyObject_MakeInstanceAttributesFromDict(PyObject *obj, PyDictOrValues *dorv)
     }
     // It's likely that this dict still shares its keys (if it was materialized
     // on request and not heavily modified):
-    assert(PyDict_CheckExact(dict));
+    if (!PyDict_CheckExact(dict)) {
+        return 0;
+    }
```

Comment:
Not something for this PR, but I think this function should either return bool (true/false) or follow the 0/-1 convention.

Reply:
I might as well do it in this PR.